std/container/array.d

```diff
@@ -231,6 +231,12 @@ private struct RangeT(A)
     }
 }
 
+extern (C) private pure @system @nogc nothrow
+{
+    pragma(mangle, "malloc") void* fakePureMalloc(size_t);
+    pragma(mangle, "realloc") void* fakePureRealloc(void* ptr, size_t size);
+}
```

Comment:
I'm hesitant to make a new module. Is there an existing module in `std.internal` that strikes you as appropriate?

Reply:
> Is there an existing module in std.internal that strikes you as appropriate?

Not really, that's why I didn't suggest a specific one.

> I'm hesitant to make a new module.

It should be okay, because the user will never see it and we can always freely reorganize things. The only annoying part is to add it to `posix.mak` and `win{32,64}.mak`

Reply:
Added `std.internal.memory`.